docs/03_goclient/15_sessions.md

```diff
@@ -0,0 +1,134 @@
+# Sessions
+
+The session framework provides a straight-forward interface for scheduling multiple activities on a single worker without having users manually specify the task list name. It also includes features like **concurrent session limitation** and **worker failure detection**.
+
+## Use Cases
+
+- **File Processing**: User may want to implement a workflow that can download a file, process it and then upload the modified version. If these three steps are implemented as three different activities, all of them should be executed by the same worker. This is a perfect use case of the session framework.
+
+- **Machine Learning Model Training**: Similar to file processing, training a machine learning model also involves three stages: download the data set, optimize the model and upload the trained parameter. However, as each model may consume a large amount of GPU memory, the number of concurrent models being trained should be limited. If the training of each model is implemented as a session, this requirement can easily be achieved by using the concurrent session limitation feature.
+
+## Basic Usage
+
+Before using the session framework to write your workflow code, your worker needs to be configured to process sessions. To do that, set the `EnableSessionWorker` field of `worker.Options` to `true` when starting your worker.
+
+The most important APIs provided by the session framework are `workflow.CreateSession()` and `workflow.CompleteSession()`. The basic idea is that all the activities executed within a session will be processed by the same worker and these two APIs allow users to create new sessions and close them after all activities finish execution.
+
+Heres a more detailed description of these two APIs:
+```go
+type SessionOptions struct {
+  // ExecutionTimeout: required, no default
+  //     Specifies the maximum amount of time the session can run
+  ExecutionTimeout time.Duration
+
+  // CreationTimeout: required, no default
+  //     Specifies how long session creation can take before returning an error
+  CreationTimeout  time.Duration
+}
+
+func CreateSession(ctx Context, sessionOptions *SessionOptions) (Context, error)
+```
+
+`CreateSession()` takes in workflow.Context, sessionOptions and returns a new session context. When it's called, it will check the task list name specified in the ActivityOptions (or the one specified in StartWorkflowOptions) and create the session on one of the workers which is polling that task list. `CreateSession()` will return an error if the context passed in already contains an open session or all the workers are currently busy and unable to handle new sessions (check the **Concurrent Session Limitation** section for more details).
```

Comment:
I find this confusing. It says that it takes workflow.Context and returns session context. I assumed that session context is a different type. I think we nee to rewrite this sentence.